### DISCUSSION TITLE
whisper touch / physical page turn buttons on kindle voyage works after restart koreader

### DISCUSSION THREAD

**Discussion by Hzj-jie** [koreader/koreader#7619] [General]
It's definitely not a new topic, but it's still annoying. After deregistering the device from amazon, I noticed a quite interesting behavior, the whisper touch works after restarting the koreader.

Before deregistering:
- start koreader
- whisper touch works
- turn off the device then turn it back on by pressing power button twice
- whisper touch does not work
- restart koreader by using menu->exit->restart
- whisper touch still does not work

After deregistering:
- start koreader
- whisper touch works
- turn off the device then turn it back on by pressing power button twice
- whisper touch does not work
- restart koreader by using menu->exit->restart
- **_whisper touch works_**

I do not know the changes amazon made during deregister, I do see a potential fix of the whisper touch for ads supported version.

A very silly but workable idea is to restart koreader after each resume. It works fine, but too silly to be considered as a solution. https://github.com/koreader/koreader/pull/7618

Instead I am thinking if resetting the inputs works. ffi/input provides only a closeAll, so for voyage, I tried to call ffi/input:closeAll, then reopen all the inputs. But unfortunately, it triggers an interesting seg fault.

```
lipc-wait-event exited normally with status: 0
[ko-input] Forked off fake event generator (pid: 20901)
Segmentation fault
lipc-wait-event exited normally with status: 0
```

I haven't worked on the ffi/input before, but likely the base/input/input.c has some code defects in `closeInputDevices`. I am pretty sure the `closeInputDevices` is designed to work as the last call, and this is a new workflow to support.
https://github.com/koreader/koreader-base/pull/1364

**Comment by NiLuJe** [comment]
Yeah, closeAll was not necessarily designed to be run at any other point than at teardown.

IIRC, the fact that whispertouch is wonky on (formerly known as SO) Ad-supported devices is very much on purpose.

Stuff *should* be working just fine on a "full" Voyage, with the latest KUAL releases.

**Reply by Hzj-jie** [reply]
Yes, I recently noticed that on my new should-be-ad-supported kindle basic 10, since I never registered it, it's in the no-ads state. koreader screensaver works fine and no annoying swipe to unlock. It gives me some guess that factory resetting or deregistering the device will move it back into the "unknown" state and work around the whisper touch.
Hopefully after https://github.com/koreader/koreader-base/pull/1364, I can get some good news.

**Reply by NiLuJe** [reply]
Sidebar for you & @hius07: I've taken down your thread on MR because this is in severe violation of the MR guidelines.

I'm also not a fan of publicizing this *anywhere*, in fact ;). Even if it is trivially deducible.

Let's not give the big A any reason to be even more annoying to deal with than they currently are.

**Reply by Hzj-jie** [reply]
LOL, sounds reasonable 🤣

**Comment by hius07** [comment]
OK, noted.
